security/fma-supervisor.md

```diff
@@ -0,0 +1,269 @@
+    - Managed Nodes may be controlled and reset by the Supervisor to maintain sync
+- (There is a different "Mode" than Managed Mode which is not yet implemented, and is described in more detail starting at FM5)
+- As Nodes A and B receive unsafe blocks over P2P, they report it to the Supervisor
+    - The Supervisor uses this to sync all the receipts to a log database
+- The Supervisor hands down the L1 block to Managed Nodes A and B
+- Each Node derives zero or more blocks from the L1 input
+- As new L2 blocks are derived from the L1, the L1:L2 link is recorded to a Derivation Database
+- As new data arrives in the Supervisor, safety promotion routines re-evaluate all given data
+    - Unsafe Heads advance as their receipts are indexed
+    - Cross Unsafe is evaluated any time new data is available
+    - Safe heads advance as their derivation is reported
+    - Cross Safe is evaluated any time new data is available
+- If *positively invalid* messages are discovered in Safe data, the Supervisor directs the given Node to Replace the block with a Deposit-Only block
+
+In this way, the Supervisor is an implementation of the Interop Protocol: it evaluates the validity of interop messages, and initiates the block replacement specified in the protocol.
+
+The Supervisor *also* serves the data it computes over RPC, for other components to check the validity of interop messages *before* they are included in a block, to avoid invalid messages.
+
+The Supervisor is as important as our Consensus Node itself — it is the primary/sole implementation of a serious aspect of the OP Spec, and if it behaves incorrectly it could mislead an entire network.
+
+# FMs:
+
+There are three conceptual ways for the Supervisor to fail:
+
+- It is unresponsive
+- It is incorrect
+- It is destructive to the Nodes connected to it in some other way
+
+And all Failure Modes are some subtype. Incorrect responses are much worse than no responses, as they may mislead the network. At maximum, a network which is mislead may have fraudulent interop messages played on it, resulting in arbitrary damage to that network. I say “mislead” because all invalid interop messages are invalid by the protocol, and this fact can be checked using L1 data. But because there is limited client diversity, bugs may have greater impact.
+
+## FM1a: Supervisor Is Totally Unavailable
+
+- Description
+    - The Supervisor is entirely unavailable, as if it were disconnected.
+    - No Message/Block Validity could be determined for any Node connected to this Supervisor.
+    - Any Node in Managed Mode would be unable to advance their chain state at all besides local-unsafe.
+    - Any Node in Standard Mode would be unable to advance cross-unsafe and cross-safe, but could still advance local-unsafe and local-safe. And because Validity can’t be checked, the Node may prefer not to trust Safe data either.
+- Risk Assessment
+    - Low Impact, High likelihood.
```

Comment:
End user impact would be similar to an unsafe head stall: the longer the supervisor is unavailable, the more $ is lost via things like changing on-chain prices making trades no longer profitable, or the delay resulting in users not being able to take time-sensitive actions. Similar to unsafe head stalls, the longer the supervisor is unavailable for, the more loss there is. Responding swiftly here is important and should be treated the same as how we respond to unsafe head stalls currently

Reply:
Just to clarify - the impact here is indeed a `cross unsafe` and beyond stall, but this would *only* be the case for for the individual nodes connected to the affected supervisor. This is not like an unsafe head stall, and rather like your `op-node` lost its L1 connection.

If the Supervisor that goes down happens to be the one a Sequencer is reliant on, the outcome would be that the block builder would stop including interop messages (because it can't validate them), but blocks would still be produced. We could call this something like an "Interop Stall", and it would indeed impact time sensitive interop activities.

I will increase the Impact here and call out the difference in affect when it's the Sequencer.